pkg/kubectl/resource/builder.go

```diff
@@ -567,25 +567,34 @@ func (b *Builder) visitorResult() *Result {
 }
 
 func (b *Builder) visitBySelector() *Result {
+	result := &Result{
```

Comment:
I don't want every exit to declare the whole struct though.  I think that's not a consistent pattern.